### PR TITLE
Use NaN in camera fields when no value known

### PR DESCRIPTION
**Pull request by peterbarker**
per-spec

Replaces https://github.com/ArduPilot/ardupilot/pull/25010/files

Creates a global `NaNf` per @rmackay9 's suggestion

### CODE REVIEW

**Comment by rmackay9** [comment]
Great to see the arrival of NANf!

**Comment by peterbarker** [comment]
There's no compiler output change right up until the changes to the mavlink packets in this PR.  Using NanF in those packets costs 16 bytes on some platforms.

**Comment by peterbarker** [comment]
I've used SITL to test the base backend, but no simulators for any of three more-specific backends

File: libraries/AP_Mount/AP_Mount_Topotek.cpp [hunk 1]
```diff
@@ -566,15 +566,13 @@ void AP_Mount_Topotek::send_camera_settings(mavlink_channel_t chan) const
...
-
     // send CAMERA_SETTINGS message
     mavlink_msg_camera_settings_send(
         chan,
         AP_HAL::millis(),   // time_boot_ms
         _recording ? CAMERA_MODE_VIDEO : CAMERA_MODE_IMAGE, // camera mode (0:image, 1:video, 2:image survey)
-        NaN,                // zoomLevel float, percentage from 0 to 100, NaN if unknown
-        NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf,                // zoomLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf);               // focusLevel float, percentage from 0 to 100, NaN if unknown
```
Comment (amilcarlucas) [inline libraries/AP_Mount/AP_Mount_Topotek.cpp:575]:
```suggestion
        NaNf,               // zoomLevel float, percentage from 0 to 100, NaN if unknown
        NaNf);              // focusLevel float, percentage from 0 to 100, NaN if unknown
```

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp [hunk 2]
```diff
@@ -951,7 +949,7 @@ void AP_Mount_Viewpro::send_camera_settings(mavlink_channel_t chan) const
         AP_HAL::millis(),   // time_boot_ms
         _recording ? CAMERA_MODE_VIDEO : CAMERA_MODE_IMAGE, // camera mode (0:image, 1:video, 2:image survey)
         zoom_level,         // zoomLevel float, percentage from 0 to 100, NaN if unknown
-        NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf);               // focusLevel float, percentage from 0 to 100, NaN if unknown
```
Comment (amilcarlucas) [inline libraries/AP_Mount/AP_Mount_Viewpro.cpp:952]:
```suggestion
        NaNf);              // focusLevel float, percentage from 0 to 100, NaN if unknown
```

File: libraries/AP_Mount/AP_Mount_Xacti.cpp [hunk 3]
```diff
@@ -375,9 +374,9 @@ void AP_Mount_Xacti::send_camera_information(mavlink_channel_t chan) const
         vendor_name,            // vendor_name uint8_t[32]
         model_name,             // model_name uint8_t[32]
         _firmware_version.received ? _firmware_version.mav_ver : 0, // firmware version uint32_t
-        NaN,                    // focal_length float (mm)
-        NaN,                    // sensor_size_h float (mm)
-        NaN,                    // sensor_size_v float (mm)
+        NaNf,                    // focal_length float (mm)
+        NaNf,                    // sensor_size_h float (mm)
+        NaNf,                    // sensor_size_v float (mm)
```
Comment (amilcarlucas) [inline libraries/AP_Mount/AP_Mount_Xacti.cpp:379]:
```suggestion
        NaNf,                   // focal_length float (mm)
        NaNf,                   // sensor_size_h float (mm)
        NaNf,                   // sensor_size_v float (mm)
```

File: libraries/AP_Mount/AP_Mount_Xacti.cpp [hunk 4]
```diff
@@ -390,15 +389,13 @@ void AP_Mount_Xacti::send_camera_information(mavlink_channel_t chan) const
...
-    const float NaN = nanf("0x4152");
-
     // send CAMERA_SETTINGS message
     mavlink_msg_camera_settings_send(
         chan,
         AP_HAL::millis(),   // time_boot_ms
         _recording_video ? CAMERA_MODE_VIDEO : CAMERA_MODE_IMAGE,   // camera mode (0:image, 1:video, 2:image survey)
         0,                  // zoomLevel float, percentage from 0 to 100, NaN if unknown
-        NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf);               // focusLevel float, percentage from 0 to 100, NaN if unknown
```
Comment (amilcarlucas) [inline libraries/AP_Mount/AP_Mount_Xacti.cpp:398]:
```suggestion
        NaNf);              // focusLevel float, percentage from 0 to 100, NaN if unknown
```

File: libraries/AP_Camera/AP_Camera_Backend.cpp [hunk 5]
```diff
@@ -241,15 +240,13 @@ void AP_Camera_Backend::send_camera_information(mavlink_channel_t chan) const
...
-
     // send CAMERA_SETTINGS message
     mavlink_msg_camera_settings_send(
         chan,
         AP_HAL::millis(),   // time_boot_ms
         CAMERA_MODE_IMAGE,  // camera mode (0:image, 1:video, 2:image survey)
-        NaN,                // zoomLevel float, percentage from 0 to 100, NaN if unknown
-        NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf,                // zoomLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf);               // focusLevel float, percentage from 0 to 100, NaN if unknown
```
Comment (amilcarlucas) [inline libraries/AP_Camera/AP_Camera_Backend.cpp:249]:
```suggestion
        NaNf,               // zoomLevel float, percentage from 0 to 100, NaN if unknown
        NaNf);              // focusLevel float, percentage from 0 to 100, NaN if unknown
```

File: libraries/AP_Camera/AP_Camera_Backend.cpp [hunk 6]
```diff
@@ -307,7 +301,7 @@ void AP_Camera_Backend::send_camera_capture_status(mavlink_channel_t chan) const
         0,                // current status of video capturing (0: idle, 1: capture in progress)
         static_cast<float>(time_interval_settings.time_interval_ms) / 1000.0, // image capture interval (s)
         0,                // elapsed time since recording started (ms)
-        NaN,              // available storage capacity (ms)
+        NaNf,              // available storage capacity (ms)
```
Comment (amilcarlucas) [inline libraries/AP_Camera/AP_Camera_Backend.cpp:304]:
```suggestion
        NaNf,             // available storage capacity (ms)
```

File: libraries/AP_Mount/AP_Mount_Siyi.cpp [hunk 7]
```diff
@@ -1120,14 +1119,13 @@ void AP_Mount_Siyi::send_camera_settings(mavlink_channel_t chan) const
         AP_HAL::millis(),   // time_boot_ms
         mode_id,            // camera mode (0:image, 1:video, 2:image survey)
         zoom_pct,           // zoomLevel float, percentage from 0 to 100, NaN if unknown
-        NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
+        NaNf);               // focusLevel float, percentage from 0 to 100, NaN if unknown
```
Comment (amilcarlucas) [inline libraries/AP_Mount/AP_Mount_Siyi.cpp:1122]:
```suggestion
        NaNf);              // focusLevel float, percentage from 0 to 100, NaN if unknown
```

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Mount/AP_Mount_Viewpro.cpp]: Fixed

File: libraries/AP_Mount/AP_Mount_Xacti.cpp (hunk 3 above)
Comment (peterbarker) [reply libraries/AP_Mount/AP_Mount_Xacti.cpp]: Fixed

File: libraries/AP_Mount/AP_Mount_Xacti.cpp (hunk 4 above)
Comment (peterbarker) [reply libraries/AP_Mount/AP_Mount_Xacti.cpp]: Fixed

File: libraries/AP_Camera/AP_Camera_Backend.cpp (hunk 5 above)
Comment (peterbarker) [reply libraries/AP_Camera/AP_Camera_Backend.cpp]: Fixed

File: libraries/AP_Mount/AP_Mount_Siyi.cpp (hunk 7 above)
Comment (peterbarker) [reply libraries/AP_Mount/AP_Mount_Siyi.cpp]: Fixed

File: libraries/AP_Camera/AP_Camera_Backend.cpp (hunk 6 above)
Comment (peterbarker) [reply libraries/AP_Camera/AP_Camera_Backend.cpp]: Fixed

File: libraries/AP_Mount/AP_Mount_Topotek.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Mount/AP_Mount_Topotek.cpp]: Fixed

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**